utils/contractParam.ts

```diff
@@ -846,25 +876,31 @@ function decodeB58(type: Expr, data: Expr): Expr {
         }
       }
       case "pair": {
-        const new_type = toRightAssociative(type);
+        toRightAssociativePairType(type);
+        data = toRightAssociativePairData(data);
+
         if ("prim" in data && !!data.args) {
           let new_data = type.args?.map((v, i) => {
+            //@ts-expect-error
             if (!data.args?.[i]) throw new Error("Internal: should have args");
+            //@ts-expect-error
             const d = decodeB58(v, data.args[i]);
             return d;
           });
           data.args = new_data;
           return data;
         } else {
-          throw new Error("Internal: data should be a prim");
+          throw new Error("Internal: data should be a prim or array");
         }
       }
       case "option": {
         if ("prim" in data) {
           if (data.prim === "Some" && !!data.args) {
             let new_data = type.args?.map((v, i) => {
+              //@ts-expect-error
```

Comment:
If I don't add this, compiler will scream...?
@JulesGuesnon  do you have an idea?

Reply:
It because your code is in a callback. As the scope is different, `data` might change, and typescript assumes the checks are only valid for a given scope. So basically, you have to do again the `Prim` check if you want to avoid `//@ts-expect-error`

Reply:
removed it.